File: rust/problem_05.rs
// Copyright 2014 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 5
// Smallest multiple

pub fn solve() -> int {
	let mut n = 20;
	loop {
		let mut good = true;
		for i in range(11, 20) {
			if n % i != 0 {
				good = false;
				break;
			}
		}
		if good {
			return n;
		}
		n += 20;
	}
}

File: c++/problem_05.cpp
// Copyright 2016 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 5
// Smallest multiple

namespace problem_05 {

int solve() {
	int n = 20;
	bool good = false;

	while (!good) {
		n += 20;
		good = true;
		for (int i = 11; i < 20; ++i) {
			if (n % i != 0) {
				good = false;
				break;
			}
		}
	}
	return n;
}

} // namespace problem_05
